packages/in-memory-live-query-store/src/throttle.ts

```diff
@@ -0,0 +1,49 @@
+import { runWith } from "./runWith";
+
+export interface ThrottledFunction {
+  run: (...args: unknown[]) => void;
+  cancel: () => void;
+}
+
+export const throttle = <T>(fn: (...args: unknown[]) => T | Promise<T>, wait: number): ThrottledFunction => {
+  let timeout: ReturnType<typeof setTimeout>;
+  let lastCalled = 0;
+  let cancelled = false;
+
+  const exec = (...args: unknown[]) => {
+    if (!cancelled) {
+      runWith(fn(...args), () => {
+        lastCalled = Date.now();
+      });
```

Comment:
Could this cause issues if we only set lastCalled after the Promise was resolved? Shouldn't we set lastCalled immediately after the function was called?